sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -1624,33 +1595,96 @@ def get_transceiver_status(self):
-            trans_status['postfecberlowwarning_flag'] = self.vdm_dict['Errored Frames Average Media Input'][1][8]
-        except KeyError:
-            pass
+        if not self.is_flat_memory():
+            dp_state_dict = self.get_datapath_state()
+            trans_status['DP1State'] = dp_state_dict['DP1State']
+            trans_status['DP2State'] = dp_state_dict['DP2State']
+            trans_status['DP3State'] = dp_state_dict['DP3State']
+            trans_status['DP4State'] = dp_state_dict['DP4State']
+            trans_status['DP5State'] = dp_state_dict['DP5State']
+            trans_status['DP6State'] = dp_state_dict['DP6State']
+            trans_status['DP7State'] = dp_state_dict['DP7State']
+            trans_status['DP8State'] = dp_state_dict['DP8State']
+            tx_output_status_dict = self.get_tx_output_status()
+            trans_status['txoutput_status'] = tx_output_status_dict['TxOutputStatus1']
+            rx_output_status_dict = self.get_rx_output_status()
+            trans_status['rxoutput_status_hostlane1'] = rx_output_status_dict['RxOutputStatus1']
+            trans_status['rxoutput_status_hostlane2'] = rx_output_status_dict['RxOutputStatus2']
+            trans_status['rxoutput_status_hostlane3'] = rx_output_status_dict['RxOutputStatus3']
+            trans_status['rxoutput_status_hostlane4'] = rx_output_status_dict['RxOutputStatus4']
+            trans_status['rxoutput_status_hostlane5'] = rx_output_status_dict['RxOutputStatus5']
+            trans_status['rxoutput_status_hostlane6'] = rx_output_status_dict['RxOutputStatus6']
+            trans_status['rxoutput_status_hostlane7'] = rx_output_status_dict['RxOutputStatus7']
+            trans_status['rxoutput_status_hostlane8'] = rx_output_status_dict['RxOutputStatus8']
+            tx_fault = self.get_tx_fault()
+            trans_status['txfault'] = tx_fault[0]
+            tx_los = self.get_tx_los()
+            trans_status['txlos_hostlane1'] = tx_los[0]
+            trans_status['txlos_hostlane2'] = tx_los[1]
+            trans_status['txlos_hostlane3'] = tx_los[2]
+            trans_status['txlos_hostlane4'] = tx_los[3]
+            trans_status['txlos_hostlane5'] = tx_los[4]
+            trans_status['txlos_hostlane6'] = tx_los[5]
+            trans_status['txlos_hostlane7'] = tx_los[6]
+            trans_status['txlos_hostlane8'] = tx_los[7]
+            tx_lol = self.get_tx_cdr_lol()
+            trans_status['txcdrlol_hostlane1'] = tx_lol[0]
+            trans_status['txcdrlol_hostlane2'] = tx_lol[1]
+            trans_status['txcdrlol_hostlane3'] = tx_lol[2]
+            trans_status['txcdrlol_hostlane4'] = tx_lol[3]
+            trans_status['txcdrlol_hostlane5'] = tx_lol[4]
+            trans_status['txcdrlol_hostlane6'] = tx_lol[5]
+            trans_status['txcdrlol_hostlane7'] = tx_lol[6]
+            trans_status['txcdrlol_hostlane8'] = tx_lol[7]
```

Comment:
can we condense these line to few lines?

Reply:
done